src/overlays/actors/ovl_En_Sekihi/z_en_sekihi.c

```diff
@@ -32,20 +36,147 @@ const ActorInit En_Sekihi_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+static s16 sObjectIds[] = { OBJECT_SEKIHIL, OBJECT_SEKIHIG, OBJECT_SEKIHIN, OBJECT_SEKIHIZ, OBJECT_ZOG };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/EnSekihi_Init.s")
+static s32 sOpaDLists[] = { &gGraveTriforceSymbolDL, &gGraveGoronSymbolDL, &gGraveKokiriSymbolDL, &gTombSongOfSoaringDL,
+                            &gMikauGraveDL };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/EnSekihi_Destroy.s")
+static s32 sXluDLists[] = { &gSunSongTriforceSymbolDL, &gSunSongGoronSymbolDL, &gSunSongKokiriSymbolDL,
+                            &gSekihizEmptyDL, &gMikauGraveDirtDL };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/func_80A44DE8.s")
+static u16 sTextIds[] = { 0, 0, 0, 0, 0x1018 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/func_80A44F40.s")
+void EnSekihi_Init(Actor* thisx, PlayState* play) {
+    EnSekihi* this = THIS;
+    s32 params = ENSIKIHI_GET_F(thisx);
+    s32 objectIndex;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/func_80A450B0.s")
+    if ((params < 0 || params >= 5) || sOpaDLists[params] == 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        if ((params == 4) && (((gSaveContext.save.skullTokenCount & 0xFFFF)) >= 0x1E)) {
+            gSaveContext.save.weekEventReg[13] |= 0x20;
+        }
+        objectIndex = Object_GetIndex(&play->objectCtx, sObjectIds[params]);
+        if (objectIndex >= 0) {
+            this->objectIndex = objectIndex;
+        }
+        this->actionFunc = func_80A44DE8;
+        this->opaDList = sOpaDLists[params];
+        this->xluDList = sXluDLists[params];
+        this->dyna.actor.textId = sTextIds[params];
+        this->dyna.actor.focus.pos.y = this->dyna.actor.world.pos.y + 60.0f;
+        Actor_SetScale(&this->dyna.actor, 0.1f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/func_80A45130.s")
+void EnSekihi_Destroy(Actor* thisx, PlayState* play) {
+    EnSekihi* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/EnSekihi_Update.s")
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/func_80A45164.s")
+void func_80A44DE8(EnSekihi* this, PlayState* play) {
+    CollisionHeader* colHeader = NULL;
+    s32 params;
+    CollisionHeader* sColHeader[] = { &gSekihilColHeader, &gSekihigColHeader, &gSekihinColHeader, &gSekihizColHeader,
+                                      &gObjectZogColHeader };
```

Comment:
```suggestion
    CollisionHeader* sColHeader[] = { &gSekihilColHeader, &gSekihigColHeader, &gSekihinColHeader, &gSekihizColHeader,
                                      &gObjectZogColHeader, };
```
Probably formats better with a trailing comma.